pkg/core/mpt/trie.go

```diff
@@ -348,14 +464,23 @@ func (t *Trie) getFromStore(h util.Uint256) (Node, error) {
 		return nil, err
 	}
 
-	var n NodeObject
+	var n nodeWrapper
 	r := io.NewBinReaderFromBuf(data)
+	if t.gcEnabled {
+		n = new(extendedNodeObject)
+	} else {
+		n = new(NodeObject)
+	}
 	n.DecodeBinary(r)
 	if r.Err != nil {
 		return nil, r.Err
 	}
-	n.Node.(flushedNode).setCache(data, h)
-	return n.Node, nil
+
+	if t.gcEnabled {
+		data = data[:len(data)-4]
```

Comment:
I think if you're to avoid this cut you can then avoid `NewBufBinWriter` in flush() and that would be really helpful.

Reply:
There is no encoding in subsequent commits. `append` in `updateRefCount` is done to this slice with a sufficient capacity.